src/GuardClauses/GuardClauseExtensions.cs

```diff
@@ -41,15 +42,16 @@ public static T Null<T>([JetBrainsNotNull] this IGuardClause guardClause, [NotNu
         /// <param name="guardClause"></param>
         /// <param name="input"></param>
         /// <param name="parameterName"></param>
+        /// <param name="message">Optional. Custom error message</param>
         /// <returns><paramref name="input" /> if the value is not an empty string or null.</returns>
         /// <exception cref="ArgumentNullException"></exception>
         /// <exception cref="ArgumentException"></exception>
-        public static string NullOrEmpty([JetBrainsNotNull] this IGuardClause guardClause, [NotNull, JetBrainsNotNull][ValidatedNotNull] string? input, [JetBrainsNotNull] string parameterName)
+        public static string NullOrEmpty([JetBrainsNotNull] this IGuardClause guardClause, [NotNull, JetBrainsNotNull][ValidatedNotNull] string? input, [JetBrainsNotNull] string parameterName, string? message = null)
         {
             Guard.Against.Null(input, parameterName);
-            if (input == String.Empty)
+            if (input == string.Empty)
             {
-                throw new ArgumentException($"Required input {parameterName} was empty.", parameterName);
+                throw new ArgumentException(message ?? $"Required input {parameterName} was empty.", parameterName);
```

Comment:
This should work fine as-is.

Reply:
Left `ArgumentException`'s in the change as is